### PR TITLE
Take Java migration classes into account in Flyway extension

### LINKED ISSUE #9819: Quarkus 1.5.0 Flyway not running Java Migrations
**Describe the bug**
Quarkus 1.5.0 Flyway not running Java Migrations

**Expected behavior**
All migrations (sql and Java) are run.

**Actual behavior**
Only the sql migrations are running.

**To Reproduce**
1. Create MySQL8 database with schema testflyway
2. Update database config in application.properties
3. Run using maven compile quarkus:dev

**Configuration**
```properties
# Add your application.properties here, if applicable.
quarkus.flyway.migrate-at-start = true
quarkus.flyway.baseline-on-migrate = true

quarkus.log.console.enable = true
%dev.quarkus.log.console.enable = true
quarkus.log.console.format = %d{yyyy-MM-dd HH\:mm\:ss,SSS} %-5p [%c{3.}] (%t) %s%e%n
quarkus.log.console.color = true
quarkus.log.console.level = DEBUG

quarkus.log.file.enable = true
quarkus.log.file.format = %d{yyyy-MM-dd HH\:mm\:ss,SSS} %-5p [%c{3.}] (%t) %s%e%n
quarkus.log.file.path = logs/app.log
quarkus.log.file.rotation.max-file-size = 10M
quarkus.log.file.rotation.max-backup-index = 5
quarkus.log.file.rotation.rotate-on-boot = true

quarkus.datasource.db-kind = mysql
quarkus.datasource.username = root
quarkus.datasource.password = xxxxx
quarkus.datasource.jdbc.url = jdbc:mysql://127.0.0.1:3306/testflyway
```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

Screen Shot 2020-06-05 at 10 59 45 PM

Screen Shot 2020-06-05 at 10 59 56 PM

Screen Shot 2020-06-05 at 11 01 22 PM

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Darwin Brents-MBP 19.5.0 Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64
 - Output of `java -version`:  OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.7+10)
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.5.0
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)

**Additional context**
(Add any other context about the problem here.)
- It works in Quarkus 1.4.2
- The sql and classes are in the target directory
- The example runs first two SQL migrations but not the Java one that should follow.

** Simple Test Project **
[testflyway.zip](https://github.com/quarkusio/quarkus/files/4735623/testflyway.zip)

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #9819

### CODE REVIEW

File: extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java [hunk 1]
```diff
@@ -87,9 +101,27 @@ void build(BuildProducer<FeatureBuildItem> featureProducer,
         List<String> applicationMigrations = discoverApplicationMigrations(getMigrationLocations(dataSourceNames));
         recorder.setApplicationMigrationFiles(applicationMigrations);
 
+        Set<Class<?>> javaMigrationClasses = new HashSet<>();
+        addJavaMigrations(combinedIndexBuildItem.getIndex().getAllKnownImplementors(JAVA_MIGRATION), context,
+                reflectiveClassProducer, javaMigrationClasses);
+        addJavaMigrations(combinedIndexBuildItem.getIndex().getAllKnownSubclasses(BASE_JAVA_MIGRATION), context,
```
Comment (gastaldi) [inline extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java:107]: `BaseJavaMigration implements JavaMigration`, wouldn't classes from the statement above be included here?

Comment (gastaldi) [reply extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java:107]: I wonder if we shouldn't just stick to the first statement

Comment (geoand) [reply extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java:107]: Unfortunately no - I tried it (unless I tried wrongi)

Comment (gastaldi) [reply extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java:107]: Perhaps the Flyway JAR need to be Jandexed too?

Comment (gastaldi) [reply extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java]: Done :wink: See if that makes sense

Comment (geoand) [reply extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java]: If the tests pass, I'm fine with it, thanks!

**Comment by geoand** [comment]
@gastaldi mind approving since you took a look anyway?

**Review by gastaldi — APPROVED** [review]
Done, good job! :+1:

**Comment by geoand** [comment]
Right back at you 🤘

**Comment by mamadaliev** [comment]
Thanks!

**Pull request merged**